pkg/security/resolvers/process/resolver.go

```diff
@@ -1185,9 +1185,9 @@ func (p *Resolver) Dump(withArgs bool) (string, error) {
 	if err != nil {
 		return "", err
 	}
-	defer dump.Close()
 
 	if err := os.Chmod(dump.Name(), 0400); err != nil {
+		dump.Close()
```

Comment:
let's keep the defer for this code path, and explicitly close at the end of the function as well like you did